core/store/migrate/migrate.go

```diff
@@ -25,20 +27,55 @@ var embedMigrations embed.FS
 
 const MIGRATIONS_DIR string = "migrations"
 
-func init() {
-	goose.SetBaseFS(embedMigrations)
-	goose.SetSequential(true)
-	goose.SetTableName("goose_migrations")
+func NewProvider(ctx context.Context, db *sql.DB) (*goose.Provider, error) {
+	store, err := database.NewStore(goose.DialectPostgres, "goose_migrations")
+	if err != nil {
+		return nil, err
+	}
+
+	goMigrations := []*goose.Migration{
+		migrations.Migration36,
+		migrations.Migration54,
+		migrations.Migration56,
+		migrations.Migration195,
+	}
+
 	logMigrations := os.Getenv("CL_LOG_SQL_MIGRATIONS")
 	verbose, _ := strconv.ParseBool(logMigrations)
-	goose.SetVerbose(verbose)
+
+	fys, err := fs.Sub(embedMigrations, MIGRATIONS_DIR)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sub filesystem for embedded migration dir: %w", err)
+	}
+	// hack to work around global go migrations
+	// https: //github.com/pressly/goose/issues/782
+	goose.ResetGlobalMigrations()
```

Comment:
any concerns of the usage of this hack?

Reply:
no, goose has already fixed the issue and it will be in a new release soon